vnext/Microsoft.ReactNative.Managed.CodeGen.UnitTests/AnalysysTestBase.cs

```diff
@@ -0,0 +1,140 @@
+{
+  public abstract class AnalysysTestBase
+  {
+    public TestContext TestContext { get; set; }
+
+    protected CancellationTokenSource TokenSource;
+
+    public AnalysysTestBase()
+    {
+      TokenSource = new CancellationTokenSource();
+    }
+
+    protected (CodeAnalyzer, INamedTypeSymbol) AnalyzeModuleFile(string csSnippet)
+    {
+      var csCode = @"
+using System;
+using Microsoft.ReactNative.Managed;
+
+[ReactModule]
+public class TestClass
+{
+" + csSnippet + @"
+}
+";
+
+      return AnalyzeFileAsync(csCode).GetAwaiter().GetResult();
+    }
+
+    protected (CodeAnalyzer, INamedTypeSymbol) AnalyzeFile(string csCode)
+    {
+      return AnalyzeFileAsync(csCode).GetAwaiter().GetResult();
+    }
+
+    protected async Task<(CodeAnalyzer, INamedTypeSymbol)> AnalyzeFileAsync(string csCode)
+    {
+      var codeAnalyzer = new CodeAnalyzer(TokenSource.Token);
+
+      codeAnalyzer.AddSyntaxTree(
+        CSharpSyntaxTree.ParseText(csCode, path: "test.cs", options: new CSharpParseOptions()));
+
+      var references = new List<string>();
+      references.AddRange(Directory.EnumerateFiles(@"C:\Users\dannyvv\.nuget\packages\microsoft.netcore.universalwindowsplatform\6.2.8\ref\uap10.0.15138", "*.dll"));
+
+      var targetFolderWithConfiguration = Path.GetDirectoryName(TestContext.TestDeploymentDir);
+      var configuration = Path.GetFileName(targetFolderWithConfiguration);
+      var targetRoot = Path.GetDirectoryName(Path.GetDirectoryName(targetFolderWithConfiguration));
+
+      // A custom target in the test project ensures the x64 versions of these assemblies are built
+      // before the unittest is run.
+      // I could not find a way for the dotnet core project to NOT be "Any CPU" and then produce x64 binaries
+      // in that configuration. I tried all the usual tricks for the netcore project to match the universal
+      // multi platform support.
+      references.Add($@"{targetRoot}\X64\{configuration}\Microsoft.ReactNative.Managed\Microsoft.ReactNative.Managed\Microsoft.ReactNative.Managed.dll");
+      references.Add($@"{targetRoot}\X64\{configuration}\Microsoft.ReactNative\Microsoft.ReactNative.winmd");
+
+      // Add the windows 10 sdk assemblies for UWP. We have a compile time safeguard in place to
+      // ensure that it matches the version that is defined in our msbuild projects matches the one
+      // used in our unittests. Unfortunately C# doesn't allow constants to flow into code, nor do any
+      // of the unitest runners are allowed to pick up a variable from the build file. And given the many
+      // ways one can run inttests, this seemed to be the most reasonable out of a lot of poor options.
+      var win10SdkFolder = @"C:\Program Files (x86)\Windows Kits\10";
+#if win10SdkVersion10_0_18362_0
+      var win10SdkVersion = "10.0.18362.0";
+#else
+  #error The Win10 Sdk Version must be updated in code when updated in msbuild.
+#endif
+
+      references.AddRange(new [] {
```

Comment:
This is usually done by the GetSDKReferences / ResolveFrameworkReferences targets?

Reply:
Yes, but since this unittest is a netcoreapp2.2 library, I can't have references to UAP stuff to have msbuild resolve it and then pick the dynamically loaded locations.
I had gone back and forth several times of the tool being a vanillay netcore app or a UWP console app and ran into significant issues both ways. I went with netcoreapp after our chat, hence the unittest needs to be netcoreapp2.2 as well.

Reply:
@dannyvv there was an email today about ADO/GHA removing older netcore support (including up to 3.0)

Reply:
I'd be happy to update, I thought we had restrictions on 2.2 due to uap from previous conversations. I'll do so as a future commit.